### PR TITLE
Documentation to move from old to new theme provider

### PR DESCRIPTION
**Pull request by rurikoaraki**
### Platforms Impacted
- [ ] iOS
- [ ] macOS
- [ ] win32 (Office)
- [ ] windows
- [ ] android

### Description of changes

In order to encourage moving off deprecated systems, I think it would be good to have a set of documents describing how to move from older to newer systems.

This document takes a stab at describing how to move from the old to new ThemeProvider.

### Verification

(how the change was tested, including both manual and automated tests)

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| Screenshot or description before this change | Screenshot or description with this change |

### Pull request checklist

This PR has considered (when applicable):
- [ ] Automated Tests
- [x] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

File: docs/pages/UpgradeGuides/ThemeProvider.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+The new ThemeProvider uses the same context as the old one, so accessing the theme still uses the same `useTheme()` hook - no functional change is needed to access the theme with the new `ThemeProvider`. The only change you'll want to make is to import `useTheme()` from the newer package:
+
+```ts
+import { useTheme } from @uifabricshared/theming-react-native;
+```
+
+to
+
+```ts
+import { useTheme } from @fluentui-react-native/theme-types;
```
Comment (ksiler) [inline docs/pages/UpgradeGuides/ThemeProvider.md:83]: I may be missing something but why are you importing useTheme from `@fluentui-react-native/theme-types` here but change it to `@fluentui-react-native/framework` in the other documentation pages?

**Comment by ksiler** [comment]
If you haven't already, it's probably worth looking at the links for Theming in the wiki (either as part of this PR or another) to make sure viewers are getting the most up-to-date information on theming and not being directed to old or stale information: https://github.com/microsoft/fluentui-react-native/wiki

**Review by ksiler — APPROVED** [review]

File: docs/pages/UpgradeGuides/ThemeProvider.md (hunk 1 above)
Comment (rurikoaraki) [reply docs/pages/UpgradeGuides/ThemeProvider.md:83]: Oh good catch, sorry, still trying to get myself into the habit of getting useTheme from framework

**Comment by rurikoaraki** [comment]
> If you haven't already, it's probably worth looking at the links for Theming in the wiki (either as part of this PR or another) to make sure viewers are getting the most up-to-date information on theming and not being directed to old or stale information: https://github.com/microsoft/fluentui-react-native/wiki

Oh thanks, I thought the index file was the main entry point, this is good to know.

**Review by FalseLobster — APPROVED** [review]

**Pull request merged**